pkg/ipcache/metadata_test.go

```diff
@@ -236,6 +236,63 @@ func TestInjectLabels(t *testing.T) {
 	option.Config.EnableIPv6 = true
 }
 
+// Test that when multiple IPs have the `resolved:host` label, we correctly
+// aggregate all labels *and* update the selector cache correctly.
+// This reproduces GH-28259.
+func TestUpdateLocalNode(t *testing.T) {
+	cancel := setupTest(t)
+	defer cancel()
+
+	ctx := context.Background()
+
+	hostid := identity.ReservedIdentityHost
+
+	bothLabels := labels.Labels{}
+	bothLabels.MergeLabels(labels.LabelHost)
+	bothLabels.MergeLabels(labels.LabelKubeAPIServer)
+
+	selectorCacheHas := func(lbls labels.Labels) {
+		t.Helper()
+		id := PolicyHandler.identities[hostid]
+		assert.NotNil(t, id)
+
+		assert.True(t, id.Equals(lbls.LabelArray()), "labels should equal: %v -- %v", id, lbls.LabelArray())
```

Comment:
When running this test locally, I struggled to understand which value that's outputted is expected vs obtained. When I modified this to 

```suggestion
		assert.Equal(t, lbls.LabelArray(), id, "labels should equal")
```

it was much easier to read.

Reply:
Somehow I thought that label arrays didn't have ordering guarantees, but that was wrong, and we can use `assert.Equal`